shell/plugins/dashboard-store/resource-class.js

```diff
@@ -910,7 +910,23 @@ export default class Resource {
     if (extensionMenuActions.length) {
       // Add a divider first
       all.push({ divider: true });
-      all = all.concat(extensionMenuActions);
+
+      extensionMenuActions.forEach((action) => {
+        const newActionInstance = { ...action };
+
+        // sets the enabled flag to true if omitted on the config
+        if (!Object.keys(newActionInstance).includes('enabled')) {
+          newActionInstance.enabled = true;
+        } else if (Object.keys(newActionInstance).includes('enabled') && typeof newActionInstance.enabled === 'function') {          
```

Comment:
The below will fix linting, improve readability, fix the max call size error and handle missconfigurations nicer

```
      extensionMenuActions.forEach((action) => {
        const newActionInstance = { ...action };

        const enabledFn = newActionInstance.enabled;
        const typeofEnabled = typeof enabledFn;

        switch (typeofEnabled) {
        case 'undefined':
          newActionInstance.enabled = true;
          break;
        case 'function':
          Object.defineProperty(newActionInstance, 'enabled', { get: () => enabledFn(this) });
          break;
        case 'boolean':
          // no op, just use it directly
          break;
        default:
          // unsupported value
          console.warn(`Unsupported 'enabled' property type for action: ${ action.label || action.labelKey }` ); // eslint-disable-line no-console
          delete newActionInstance.enabled;
          break;
        }

        all.push(newActionInstance);
      });
```

Reply:
Thanks @richard-cox ! 🙏 🙇